packages/framework/src/Model/AdvancedSearch/Filter/ProductFlagFilter.php

```diff
@@ -74,7 +74,8 @@ public function extendQueryBuilder(QueryBuilder $queryBuilder, $rulesData)
             if ($ruleData->operator === self::OPERATOR_IS) {
                 $tableAlias = 'f' . $index;
                 $flagParameter = 'flag' . $index;
-                $queryBuilder->join('p.flags', $tableAlias, Join::WITH, $tableAlias . '.id = :' . $flagParameter);
+                $queryBuilder->join(ProductDomain::class, 'pd', Join::WITH, 'p.id = pd.product');
```

Comment:
I'm not sure how to handle this properly, but currently filtering by flag duplicates the data in grid (it's due to joining table without the domain id). 

I think we want to include products that have a flag on any domain, but do not duplicate them. 

<img width="1417" alt="Snímek obrazovky 2024-01-18 v 14 59 25" src="https://github.com/shopsys/shopsys/assets/1177414/c846c9d7-b038-4ea2-b658-8fa4c3e57b9c">

Reply:
I have reworked it